### PR TITLE
chore(react): allow components to be imported through subpath

### PR DESCRIPTION
**Pull request by thivi**
### Purpose
This PR adds scripts to allow components to be imported from their subpaths.
This makes importing like this possible
```TypeScript
import Button from "@oxygen-ui/react/Button"
```

### Related Issues
- Issue `#1` or (None)

### Related PRs
- Related PR `#1` or (None)

### Checklist
- [ ] UX/UI review done on the final implementation.
- [ ] Story provided. (Add screenshots)
- [ ] Manual test round performed and verified.
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Documentation provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: packages/react/scripts/build-components.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+const outputDir = path.resolve(__dirname, '..', 'dist', 'transpiled');
+
+logger.log('💅 Transpiling components...');
+execSync(
+  `npx tsc --project tsconfig.lib.json --declarationDir ${outputDir} --emitDeclarationOnly false --outDir ${outputDir}`,
+);
+logger.log('💅 Components transpiled! 🎉🎉🎉');
+
+logger.log('💅 Copying components...');
+fs.rmSync(path.resolve(__dirname, '..', 'dist/transpiled/src/components/index.js'));
```
Comment (brionmario) [inline packages/react/scripts/build-components.js:35]: Shall we use ` path.join`?
```suggestion
fs.rmSync(path.resolve(__dirname, '..', path.join('dist', 'transpiled', 'src', 'components', 'index.js'));
```

File: packages/react/scripts/fix-files.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      !moduleDirs.includes(file)
+    );
+  });
+
+  /**
+   * This method fixes the references to the non-component directories in the dist folder such as utils, constants, etc.
+   * in imports.
+   *
+   * @param {string} file - The file to be fixed.
+   *
```
Comment (brionmario) [inline packages/react/scripts/fix-files.js:51]: Do we really need a space here. https://tsdoc.org/pages/tags/returns/

File: packages/react/scripts/fix-files.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+   * @returns {string} - The fixed file.
+   */
+  const fixImportStatements = file => {
+    const regExpString = `(import.*from.*[\"|'])(.*(${nonComponentDirsInDist.join('|')}).*)([\"|'].*)`;
+    const regExp = new RegExp(regExpString, 'g');
+    return file.replace(regExp, (_fullMatch, groupOne, groupTwo, _groupThree, groupFour) => {
```
Comment (brionmario) [inline packages/react/scripts/fix-files.js:57]:
```suggestion

    return file.replace(regExp, (_fullMatch, groupOne, groupTwo, _groupThree, groupFour) => {
```

Comment (thivi) [reply packages/react/scripts/fix-files.js:58]: Resolved by https://github.com/wso2/oxygen-ui/pull/67/commits/0e01ce8265da34cda086c64e374195b9cc0684a4

File: packages/react/scripts/fix-files.js (hunk 2 above)
Comment (thivi) [reply packages/react/scripts/fix-files.js]: Resolved by https://github.com/wso2/oxygen-ui/pull/67/commits/0e01ce8265da34cda086c64e374195b9cc0684a4

**Review by brionmario — DISMISSED** [review]

**Review by brionmario — APPROVED** [review]

**Pull request merged**